### PR TITLE
feature: tauri app

### LINKED ISSUE #48: Feature: DesktopApp with Tauri
**Is your feature request related to a problem? Please describe.**
Would you be open to create a desktop app using [tauri](https://tauri.app/) ?
It wont add to much work or maintenance overall!
And i can setup it if you agree

### PR DESCRIPTION
**Pull request by leon3s**
Closes #48
It's not working on linux currently.
There is an issue on it [there](https://github.com/tauri-apps/tauri/issues/2600)
it should work for macos and windows, but i couldn't test it..

You need the tauri cli to test:
```sh
cargo install tauri-cli
cargo tauri dev
```

The yew-app and actix-api have to be started before.

image

### CODE REVIEW

**Comment by griffobeid** [comment]
This is a great first start, and thank you for your efforts on this! 

I'm testing on MacOS and connecting to the server is working, heartbeats are being sent and received. I think that there is a lack of support on the web view that Tauri is using for audio and video transcoders so the Tauri app is unable to show incoming audio/video and is also not sending audio/video to the other users. Do you know if it is possible to use Chrome as the underlying web view with Tauri?

**Comment by darioalessandro** [comment]
@leon3s This awesome!!! thanks for the contribution!!

**Comment by leon3s** [comment]
> Do you know if it is possible to use Chrome as the underlying web view with Tauri?
I pretty sure you can't.
The goal of tauri is the use the current webkit of the system. Unlike electron that embed chomium.
That way the final binary is smaller.

Did you tested @darioalessandro ? I'm pretty sure it would work on recent version of Mac!
I'll be able to test on windows soon!

**Comment by leon3s** [comment]
What's up ?

Screenshot 2023-06-11 194442

Looks like it's working well on window!

**Comment by darioalessandro** [comment]
Testing now

**Comment by darioalessandro** [comment]
This lgtm. 

I do think we need to show an error for older browsers or supports that do not support webcodecs in general, I'll add a ticket to the backlog for that.

**Comment by darioalessandro** [comment]
https://github.com/security-union/zoom-rs/issues/56

**Review by darioalessandro — APPROVED** [review]

File: src-tauri/info.plist [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+	<key>NSCameraUsageDescription</key>
+	<string>Request camera access</string>
+	<key>NSMicrophoneUsageDescription</key>
+	<string>Request microphone access</string>
```
Comment (darioalessandro) [inline src-tauri/info.plist:7]: This goes on a different PR but on MacOS we can use `LSMinimumSystemVersion` to 13.4

**Comment by darioalessandro** [comment]
Screenshot 2023-06-11 at 5 56 29 PM
Even the latest MacOS is missing some classes

**Comment by griffobeid** [comment]
We are using some apis that safari does not support, audio encoders and media stream track processors. I think we can merge this and work on those issues separately though

**Comment by darioalessandro** [comment]
@leon3s seems like AVVideoCaptureSource is not present in Safari, I am running the latest version of MacOS. 

I think that we can use electron instead of Tauri for MacOS while they fix Safari.

**Pull request merged**